docker/scripts/install_R_version

```diff
@@ -58,103 +65,329 @@ chmod a+x /usr/bin/activate_R_${R_VERSION}
+}
+
+
+function getAndInstallFromCRAN(){
+  set +x
+  NAME="$1"
+  VERSION="$2"
+  getAndInstall "https://cran.r-project.org/src/contrib/Archive/${NAME}/${NAME}_${VERSION}.tar.gz" "$NAME" "$VERSION" || \
+  getAndInstall "https://cran.r-project.org/src/contrib/${NAME}_${VERSION}.tar.gz" "$NAME" "$VERSION" || exit 1
+  set -x
+}
+
+# How to get dependencies?
+#=========================
+# Since we can't rely on `versions` package to get older packages (Microsoft's CRAN (MRAN) is discontinued), we have to
+# install the packages in some other way. What I did was opened the dev-r-3.3.3-jdk-8 and got the installed packages by
+# using the following:
+#
+########################################################################################################################
+# # Get packages and freeze them
+# write.csv(installed.packages(), "pkgs.csv")
+########################################################################################################################
+#
+# Now that I got the packages, their versions and their dependencies, I am able to topologically sort them to ensure
+# I won't try to install a package that has is missing some dependencies. Then I install it from CRAN using the above
+# defined shell function `getAndInstallFromCRAN`.
+#
+# To make it repeatable I created the following script in R that uses `igraph` package for creating the graph representing
+# packages and its dependencies that are then topologically sorted and printed out so it can be easily copy-pasted here.
+# Some packages may never reach CRAN and those should be added to the exclude_pkgs vector (like the `isofor` package)
+# and then installed separately.
+#
+########################################################################################################################
+# exclude_pkgs <- c("isofor")
+#
+# # Load them, sort them topologically and generate the code to install them
+# pkgs <- read.csv("/Users/tomasfryda/tmp/jenkins_logs/pkgs.csv")
+# pkgs <- pkgs[pkgs$Priority != "base" | is.na(pkgs$Priority),]
+# rownames(pkgs) <- pkgs$Package
+#
+# library(igraph)
+#
+# i2n <- setNames(pkgs$Package, seq_len(nrow(pkgs)))
+# n2i <- setNames(names(i2n), i2n)
+#
+# g <- make_empty_graph(directed = TRUE)
+# g <- add_vertices(g, nrow(pkgs), label = i2n)
+#
+# for (n in i2n) {
+#   deps <- paste0(pkgs[n, "Depends"], ", ", pkgs[n, "Imports"])
+#   cat(n, "->", deps, "\n")
+#   if (is.na(deps))
+#     next
+#   deps <- strsplit(deps, ",\\s*", perl = TRUE)[[1]]
+#   for (d in deps) {
+#     print(d)
+#     d <- strsplit(d, "\\s+", perl = TRUE)[[1]][[1]]
+#     print(d)
+#     if (d %in% i2n) {
+#       g <- add_edges(g, c(n2i[[n]], n2i[[d]]))
+#     }
+#   }
+# }
+#
+# ts <- topo_sort(g, mode = "in")
+# ts <- ts[!ts %in% n2i[exclude_pkgs]]
+# cat(paste("getAndInstallFromCRAN", i2n[ts], pkgs[i2n[ts], "Version"]), sep = "\n")
+########################################################################################################################
```

Comment:
MRAN doesn't work anymore so I needed some other way of installing old R packages. Even installing `devtools` is very hard for old R versions as some packages are not available for the old versions so I decided to do equivalent of `pip freeze` and manually resolve the dependencies and order the packages in such a way that they are installed in correct order. The little R script I have in the comment is responsible for sorting the packages in the correct order.